lib/rules.js

```diff
@@ -0,0 +1,241 @@
+  );
+  const map = mapFromString(str);
+  MapRule.call(this, map, opts);
+}
+inherits(RemoveRule, MapRule);
+
+// Define several general rules that may apply to strings from any language.
+
+// Identity rule is a noop. It is always the last rule applied in every
+// Equivalence instance.
+const identity = (s1, s2) => [s1, s2];
+const identityRule = new FunctionRule(identity, { name: 'identity' });
+
+// Capitalization rule lowercases its inputs.
+const capitalization = (s1, s2) => {
+  return [s1.toLowerCase(), s2.toLowerCase()];
+};
+const capitalizationRule = new FunctionRule(capitalization, {
+  name: 'capitalization',
+});
+
+// Unicode normaliation rule transforms inputs to NFC form.
+const unicodeNormalization = (s1, s2) => {
+  return [normalize(s1, 'NFC'), normalize(s2, 'NFC')];
+};
+const unicodeNormalizationRule = new FunctionRule(unicodeNormalization, {
+  name: 'unicodeNormalization',
+});
+
+// Whitespace differences rule transforms all sequences of whitespaces to a
+// single SPACE.
+const RE_ANY_WHITESPACE = /[\s]+/g;
+const whitespaceDifferences = (s1, s2) => {
+  return [
+    s1.trim().replace(RE_ANY_WHITESPACE, ' '),
+    s2.trim().replace(RE_ANY_WHITESPACE, ' '),
+  ];
+};
+const whitespaceDifferencesRule = new FunctionRule(whitespaceDifferences, {
+  name: 'whitespaceDifferences',
+});
+
+// The following rules transform various diacritical characters, particluar
+// those that are relevant to English and Spanish words.
+const acuteAccentCombining = '\u0301';
+// Note that Unicode calls this "diaresis", see
+// https://en.wikipedia.org/wiki/Diaeresis_(diacritic)
+const umlautCombining = '\u0308';
+
+// https://www.fileformat.info/info/unicode/block/combining_diacritical_marks/list.htm
+// U+0300–U+036F
+const startOfCombiningDiacriticsBlock = 0x300;
+const endOfCombiningDiacriticsBlock = 0x36f;
+
+// Array.prototype.fill and String.prototype.repeat aren't available in IE 11.
+const arr = [];
+for (
+  let count = 0;
+  count < endOfCombiningDiacriticsBlock - startOfCombiningDiacriticsBlock;
+  count++
+) {
+  arr.push(null);
+}
+const combiningDiacriticsBlock = arr.map((_, idx) =>
+  String.fromCodePoint(idx + startOfCombiningDiacriticsBlock)
+);
+
+const commonCombiningDiacritics = [
+  '\u0300', // grave accent
+  acuteAccentCombining, // acute accent
+  '\u0302', // circumflex
+  '\u0303', // tilde
+  umlautCombining, // diaresis, aka umlaut
+  '\u0327', // cedilla
+];
+
+// Generate a remove rule from a list of codepoints to be removed.
+const codepointRuleGenerator = (codepoints, name) => {
+  return new FunctionRule(
+    (s1, s2) => {
+      let s1prime = normalize(s1, 'NFD');
+      let s2prime = normalize(s2, 'NFD');
+
+      codepoints.forEach(c => {
+        s1prime = replaceAll(s1prime, c);
+        s2prime = replaceAll(s2prime, c);
+      });
+
+      // FIXME: should be restored to whatever normalization they came in with.
+      return [normalize(s1prime, 'NFC'), normalize(s2prime, 'NFC')];
```

Comment:
What's this comment about? Did you want to do this as a part of this PR?

Reply:
It's a subtle problem that doesn't affect any of our current uses cases at all, so I'm good with deferring it.